rotkehlchen/globaldb/handler.py

```diff
@@ -1011,6 +1011,123 @@ def add_historical_prices(entries: List['HistoricalPrice']) -> None:
 
         connection.commit()
 
+    @staticmethod
+    def add_single_historical_price(entry: HistoricalPrice) -> None:
+        """Adds the given historical price entries in the DB
+
+        If any addition causes a DB error it's skipped and an error is logged
+        """
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        try:
+            serialized = entry.serialize_for_db()
+            cursor.execute(
+                'SELECT COUNT(*) FROM price_history WHERE from_asset=? AND '
+                'to_asset=? AND timestamp=?',
+                (serialized[0], serialized[1], serialized[3]),
+            )
+            if cursor.fetchone()[0] is None:
+                return
+            cursor.execute(
+                """INSERT OR IGNORE INTO price_history(
+                from_asset, to_asset, source_type, timestamp, price
+                ) VALUES (?, ?, ?, ?, ?)
+                """,
+                serialized,
+            )
+        except sqlite3.IntegrityError as e:
+            connection.rollback()  # roll back any of the executemany that may have gone in
+            log.error(
+                f'One of the given historical price entries caused a DB error. {str(e)}. ',
+            )
+        connection.commit()
+
+    @staticmethod
+    def get_manual_prices(asset: 'Asset') -> List[Dict[str, str]]:
+        """Returns prices added to the database by the user for an asset"""
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        querystr = (
+            'SELECT from_asset, to_asset, source_type, timestamp, price FROM price_history '
+            'WHERE from_asset=? AND source_type="A" ORDER BY timestamp'
+        )
+        query = cursor.execute(querystr, (asset.identifier,))
+        result = query.fetchall()
+        return [
+            {
+                'from_asset': entry[0],
+                'to_asset': entry[1],
+                'timestamp': entry[3],
+                'price': entry[4],
+            }
+            for entry in result
+        ]
+
+    def edit_manual_price(self, entry: HistoricalPrice) -> bool:
+        """Returns prices added to the database by the user for an asset"""
```

Comment:
Copy pasta docstring. Please write a proper docstring. For example explain that what you are doing here is that if the entry does not exist, you are actually adding it. Which is not good. You should not do that.